### PR TITLE
New implementations with new math contexts for Pow and PowBigInt func…

### PR DESCRIPTION
**Pull request by alexeykiselev**
…tions. New test on corner cases. New benchmarks of Pow function.

Old implementation of Pow function preserved as PowV1 for backward compatibility. Test on backward compatibility added.

### CODE REVIEW

File: pkg/ride/math/math.go [hunk 1]
```diff
@@ -43,53 +112,84 @@ func convertToBigIntResult(v *decimal.Big, scale int, mode decimal.RoundingMode)
...
+	if !checkScales(baseScale, exponentScale, resultScale) {
+		return 0, errors.New("invalid scale")
+	}
+	b := decimal.WithContext(longContext).SetMantScale(base, baseScale)
+	e := decimal.WithContext(longContext).SetMantScale(exponent, exponentScale)
+	r, err := pow(b, e, longContext)
+	if err != nil {
+		return 0, err
+	}
+	r, err = rescale(r, resultScale, 19, longContext)
```
Comment (nickeskov) [inline pkg/ride/math/math.go:157]: I think using of `longContext.Precision` would be better than `19`.

File: pkg/ride/math/math.go [hunk 2]
```diff
@@ -43,53 +112,84 @@ func convertToBigIntResult(v *decimal.Big, scale int, mode decimal.RoundingMode)
...
-	e := decimal.WithContext(decimal.Context128).SetBigMantScale(exponent, exponentScale)
-	r, err := pow(b, e)
+	b := decimal.WithContext(bigIntContext).SetBigMantScale(base, baseScale)
+	e := decimal.WithContext(bigIntContext).SetBigMantScale(exponent, exponentScale)
+	r, err := pow(b, e, bigIntContext)
 	if err != nil {
 		return nil, err
 	}
-	return convertToBigIntResult(r, resultScale, mode)
+	r, err = rescale(r, resultScale, 154, bigIntContext)
```
Comment (nickeskov) [inline pkg/ride/math/math.go:181]: I think using of `bigIntContext.Precision` would be better than `154`.

File: pkg/ride/math/math.go [hunk 3]
```diff
@@ -43,53 +112,84 @@ func convertToBigIntResult(v *decimal.Big, scale int, mode decimal.RoundingMode)
...
+		return nil, err
+	}
+	if !r.IsNormal() {
+		return nil, errors.New("not normal")
+	}
+	context := bigIntContext
+	context.RoundingMode = mode
+	r = decimal.WithContext(context).Set(r)
+	return r.RoundToInt().Int(nil), nil
+	//return convertToBigIntResult(r, resultScale, mode)
```
Comment (nickeskov) [inline pkg/ride/math/math.go:192]: Unnecessary comment.
